store/rootmulti/store.go

```diff
@@ -1076,12 +1082,10 @@ func commitStores(version int64, storeMap map[types.StoreKey]types.CommitKVStore
 			continue
 		}
 
-		if !removalMap[key] {
```

Comment:
Also including commit info of store deletion allows us to pass `TestSetLoader` in x/upgrades. The test require us to load the deleted store at the height it got deleted, If there's no commit info of the deleted store at its deletion height, we'll got store height mismatch error. This is because `loadStore(ver)` fetches the `commit infos` at height `ver` to decide what version to load for each of the module stores, usually the version to load for a module store will be equal to `ver` but if the commit info of a store is not found in `commit infos`, the store will be loaded at version 0. This lead to mismatch height problem

Reply:
@tac0turtle, @alexanderbez  What do you guys think ?

Reply:
I agree. It seems prudent to keep this logic as is

Reply:
> LGTM apart from the `commitStores` change.

I just revert the changes to `commitStores`, I also have to remove a check from `TestSetLoader` since It won't pass with the current `commitStores`

Reply:
is this good to merge now?